addons/udes_stock/models/stock_picking.py

```diff
@@ -205,23 +206,15 @@ def action_cancel(self):
         Check if picking batch is now complete
         """
         res = super(StockPicking, self).action_cancel()
-        self.check_batch()
         return res
 
-    @api.one
-    def check_batch(self):
-        """ If picking state changes, check if batch is now complete
-        """
-        if self.batch_id:
-            self.batch_id.check_batches()
-
     def write(self, vals):
         """ If writing picking, check if previous batch is now complete
         """
         batches = self.mapped(lambda p: p.batch_id)
-        res = super(StockPicking, self).write(vals)
-        batches.check_batches()
-        return res
+        context_vals = {'orig_batches': batches} if batches else {}
```

Comment:
I think a comment about `recompute` would be nice, as I think you're actually sending a message there, right?